include/fuzzuf/optimizer/pso.hpp

```diff
@@ -0,0 +1,229 @@
+/*
+ * fuzzuf
+ * Copyright (C) 2022 Ricerca Security
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see http://www.gnu.org/licenses/.
+ */
+
+#ifndef FUZZUF_INCLUDE_OPTIMIZER_PSO_HPP
+#define FUZZUF_INCLUDE_OPTIMIZER_PSO_HPP
+
+#include <algorithm>
+#include <array>
+#include <cstdint>
+#include <functional>
+#include <random>
+
+#include "fuzzuf/optimizer/optimizer.hpp"
+#include "fuzzuf/optimizer/pso.hpp"
+#include "fuzzuf/optimizer/store.hpp"
+#include "fuzzuf/utils/common.hpp"
+#include "fuzzuf/utils/random.hpp"
+
+// for Particle Swarm Optimization
+
+namespace fuzzuf::optimizer {
+
+namespace keys {}
+
+template <size_t Dimension, size_t ParticleNum>
+class PSO;
+
+template <size_t Dimension>
+class Particle {
+ public:
+  Particle();
+  ~Particle();
+
+  template <size_t _Dimension, size_t ParticleNum>
+  friend class PSO;
+
+ protected:
+  std::array<double, Dimension> position;
+  double fitness;
+  std::array<double, Dimension> velocity;
+  std::array<double, Dimension> best_position;
+  double best_fitness;
+};
+
+template <size_t Dimension, size_t ParticleNum>
```

Comment:
If you will retain this file, then I have to review this class and request changes.
But the changes would be somewhat not small.
So considering the class is (and almost certainly will be) not used,  you can just comment something like "FIXME: bababa..." instead of applying changes if you don't want to take time.

Reply:
I believe all issues are resolved in b6207e50d21b625fb526c646059baf36fe5a43c4 (and I think this class would not be used as well)